PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamPairCleaner.cxx

```diff
@@ -202,6 +202,51 @@ void AliFemtoDreamPairCleaner::CleanDecayInvMass(std::vector<AliFemtoDreamBasePa
   if (!fMinimalBooking)
     fHists->FillDaughtersSharedDaughter(histnumber, counter);
 }
+void AliFemtoDreamPairCleaner::CleanDecayAtRandom(std::vector<AliFemtoDreamBasePart> *Decay, int histnumber)
+{
+  int counter = 0;
+  Double_t randomNumber = gRandom->Rndm();
```

Comment:
I would use Uniform(), Further, this uses the same random number now for any pair. In order to sample it only in case it is necessary, I would just directly do it in the loop: if (gRandom->Uniform(0,1) < 0.5) { ... } else { ... }